src/vivarium_csu_ltbi/components/treatment.py

```diff
@@ -0,0 +1,165 @@
+                                                SUSCEPTIBLE_TB_POSITIVE_HIV)
+
+
+# noinspection PyAttributeOutsideInit
+class LTBITreatmentCoverage:
+
+    @property
+    def name(self):
+        return 'ltbi_treatment_coverage'
+
+    def setup(self, builder):
+        self.time = builder.time.clock()
+
+        self.treatment_stream = builder.randomness.get_stream(f'{self.name}.treatment_selection')
+        self.adherence_stream = builder.randomness.get_stream(f'{self.name}.adherence_propensity')
+
+        self.household_tb_exposure = builder.value.get_value('household_tuberculosis.exposure')
+
+        # NOTE: Just write these into the same key, later
+        three_hp_adherence_data = builder.data.load("three_hp.adherence")
+        six_h_adherence_data = builder.data.load("six_h.adherence")
+        self.adherence = builder.lookup.build_table(pd.concat([three_hp_adherence_data, six_h_adherence_data], axis=0),
+                                                    parameter_columns=['age', 'year'],
+                                                    key_columns=['sex', 'treatment_type'],
+                                                    value_columns=['value'])
+
+        six_h_coverage_data = builder.data.load("six_h.coverage.proportion")
+        self.six_h_with_hiv, self.six_h_under_five_hhtb = self.setup_coverage_tables(builder, six_h_coverage_data)
+        three_hp_coverage_data = builder.data.load("three_hp.coverage.proportion")
+        self.three_hp_with_hiv, self.three_hp_under_five_hhtb = self.setup_coverage_tables(builder,
+                                                                                           three_hp_coverage_data)
+
+        self.coverage = builder.value.register_value_producer('ltbi_treatment.coverage', source=self.get_coverage,
+                                                              preferred_post_processor=self.enforce_not_eligible)
+
+        self._treatment_status = pd.Series()
+        self.treatment_status = builder.value.register_value_producer('treatment_status.category',
+                                                                      source=lambda index:
+                                                                      self._treatment_status[index])
+
+        self.columns_created = ['treatment_date', 'treatment_type', 'adherence_propensity']
+        builder.population.initializes_simulants(self.on_initialize_simulants,
+                                                 requires_columns=[],
+                                                 creates_columns=self.columns_created)
+
+        self.disease_state_column = "tuberculosis_and_hiv"
+        self.population_view = builder.population.get_view([self.disease_state_column, 'age', 'sex', 'alive'] +
+                                                           self.columns_created,
+                                                           query="alive == 'alive' & treatment_type == 'untreated'")
```

Comment:
I put the filter untreated query here, but I think that is less clear than using it with get so I will move it @collijk

Reply:
Oh, yeah, didn't see that.